docs/source/sft_trainer.md

```diff
@@ -1,568 +1,293 @@
+To train on assistant messages only, use a [conversational](dataset_formats#conversational) dataset and set `assistant_only_loss=True` in the [`SFTConfig`]. This setting ensures that loss is computed **only** on the assistant responses, ignoring user or system messages.
 
 ```python
-def formatting_func(example):
-    text = f"### Question: {example['question']}\n ### Answer: {example['answer']}"
-    return text
-
-training_args = SFTConfig(packing=True)
-trainer = SFTTrainer(
-    "facebook/opt-350m",
-    train_dataset=dataset,
-    args=training_args,
-    formatting_func=formatting_func
-)
-
-trainer.train()
+training_args = SFTConfig(assistant_only_loss=True)
 ```
 
-### Control over the pretrained model
+![train_on_assistant](https://huggingface.co/datasets/trl-lib/documentation-images/resolve/main/train_on_assistant.png)
 
-You can directly pass the kwargs of the `from_pretrained()` method to the [`SFTConfig`]. For example, if you want to load a model in a different precision, analogous to
+<Tip warning={true}>
+This functionality is only available for chat templates that support returning the assistant tokens mask via the `{% generation %}` and `{% endgeneration %}` keywords. For an example of such an template, see [HugggingFaceTB/SmolLM3-3B](https://huggingface.co/HuggingFaceTB/SmolLM3-3B/blob/main/chat_template.jinja#L76-L82).
+</Tip>
 
-```python
-model = AutoModelForCausalLM.from_pretrained("facebook/opt-350m", torch_dtype=torch.bfloat16)
+### Train on completion only
 
-...
+To train on completion only, use a [prompt-completion](dataset_formats#prompt-completion) dataset. By default, the trainer computes the loss on the completion tokens only, ignoring the prompt tokens. If you want to train on the full sequence, set `completion_only_loss=False` in the [`SFTConfig`].
 
-training_args = SFTConfig(
-    model_init_kwargs={
-        "torch_dtype": "bfloat16",
-    },
-    output_dir="/tmp",
-)
-trainer = SFTTrainer(
-    "facebook/opt-350m",
-    train_dataset=dataset,
-    args=training_args,
-)
+![train_on_completion](https://huggingface.co/datasets/trl-lib/documentation-images/resolve/main/train_on_completion.png)
 
-trainer.train()
-```
-Note that all keyword arguments of `from_pretrained()` are supported.
+<Tip>
+Training on completion only is compatible with training on assistant messages only. In this case, use a [conversational](dataset_formats#conversational) [prompt-completion](dataset_formats#prompt-completion) dataset and set `assistant_only_loss=True` in the [`SFTConfig`].
+</Tip>
 
-### Training adapters
+### Train adapters with PEFT
 
-We also support tight integration with 🤗 PEFT library so that any user can conveniently train adapters and share them on the Hub instead of training the entire model.
+We support tight integration with 🤗 PEFT library so that any user can conveniently train adapters and share them on the Hub instead of training the entire model.
```

Comment:
```suggestion
We support tight integration with 🤗 PEFT library, allowing any user to conveniently train adapters and share them on the Hub, rather than training the entire model.
```